share/completions/p4.fish

```diff
@@ -0,0 +1,803 @@
+# Fish completions for Perforce (p4) SCM
+# Based on the list of commands from
+# https://www.perforce.com/perforce/r16.1/manuals/cmdref/
+# Author: Boris Aranovich https://github.com/nomaed
+
+#########################################################
+### p4 command parsing
+#########################################################
+
+function __fish_print_p4_client_name
+    set -l p4info (p4 info -s 2> /dev/null)
+    if test -z "$p4info"
```

Comment:
These `test -z`s aren't necessary because you're only doing `string` stuff later.